File: code-samples/python/r-offer-check.py
#!/usr/bin/env python3
"""
KwikAPI SDK — R-Offer Check
─────────────────────────────────────────────────────────────────────────────
Checks for active R-Offers (retention/special offers) running on a prepaid mobile number. R-Offers are personalised discount plans offered by the operator to retain subscribers. Currently supported only for Airtel and VI (Vodafone Idea) networks in India. Always check network before calling — will return an error for unsupported operators.

Endpoint  : POST /api/v2/R-OFFER_check.php
Group     : Fetching APIs
Rate Limit: Per account
    # NOTE: Supported networks: Airtel and VI (Vodafone Idea) only.

Requirements:
    pip install requests

Environment:
    UAT (testing) : https://uat.kwikapi.com
    Production    : https://www.kwikapi.com  (change BASE_URL)

Get your API key: https://kwikapi.com/auth/register
"""

import requests
from typing import Any

BASE_URL = "https://uat.kwikapi.com"  # Switch to https://www.kwikapi.com for production
API_KEY  = "YOUR_API_KEY"


class KwikAPIError(Exception):
    """Raised when the KwikAPI returns a non-success response."""


def r_offer_check(api_key: str = 'YOUR_API_KEY', opid: str = 'OPERATOR_ID', mobile: str = '9999999999') -> dict[str, Any]:
    """
    R-Offer Check

    Args:
        api_key (string): Required. Your KwikAPI API key
        opid (int): Required. Operator ID — Airtel or VI only
        mobile (string): Required. 10-digit prepaid mobile number to check offers for

    Returns:
        dict: Parsed JSON response from KwikAPI.

    Raises:
        KwikAPIError: If the API returns success=false.
        requests.HTTPError: On non-2xx HTTP status.
    """
    url = BASE_URL + "/api/v2/R-OFFER_check.php"

    with requests.Session() as session:
        session.headers.update({"Accept": "application/json"})
        response = session.post(url, data={'api_key': api_key, 'opid': opid, 'mobile': mobile})

    response.raise_for_status()
    data = response.json()

    if not data.get("success"):
        raise KwikAPIError(data.get("message", "Unknown API error"))

    return data


# ── Example usage ─────────────────────────────────────────────────────────────
if __name__ == "__main__":
    try:
        result = r_offer_check('YOUR_API_KEY', 'OPERATOR_ID', '9999999999')
        import json
        print(json.dumps(result, indent=2))
    except KwikAPIError as e:
        print(f"API Error: {e}")
        raise SystemExit(1)
    except requests.RequestException as e:
        print(f"Request failed: {e}")
        raise SystemExit(1)

File: code-samples/php/r-offer-check.php
<?php
/**
 * KwikAPI SDK — R-Offer Check
 * ─────────────────────────────────────────────────────────────────────────────
 * Checks for active R-Offers (retention/special offers) running on a prepaid mobile number. R-Offers are personalised discount plans offered by the operator to retain subscribers. Currently supported only for Airtel and VI (Vodafone Idea) networks in India. Always check network before calling — will return an error for unsupported operators.
 *
 * Endpoint  : POST /api/v2/R-OFFER_check.php
 * Group     : Fetching APIs
 * Rate Limit: Per account
 * NOTE: Supported networks: Airtel and VI (Vodafone Idea) only.
 *
 * Parameters:
 *   @param  string   $api_key          (required)  Your KwikAPI API key
 *   @param  int      $opid             (required)  Operator ID — Airtel or VI only
 *   @param  string   $mobile           (required)  10-digit prepaid mobile number to check offers for
 *
 * @return array  Decoded JSON response from KwikAPI
 * @throws RuntimeException  on cURL failure or invalid JSON response
 *
 * Environment:
 *   UAT (testing) : https://uat.kwikapi.com
 *   Production    : https://www.kwikapi.com  (change KWIKAPI_BASE_URL)
 *
 * @link https://kwikapi.com/auth/register  Get your API key
 */

define('KWIKAPI_BASE_URL', 'https://uat.kwikapi.com'); // Switch to https://www.kwikapi.com for production
define('KWIKAPI_KEY', 'YOUR_API_KEY');

/**
 * R-Offer Check
 *
 *   @param  string   $api_key          (required)  Your KwikAPI API key
 *   @param  int      $opid             (required)  Operator ID — Airtel or VI only
 *   @param  string   $mobile           (required)  10-digit prepaid mobile number to check offers for
 * @return array
 * @throws RuntimeException
 */
function kwik_r_offer_check(string $api_key = 'YOUR_API_KEY', string $opid = 'OPERATOR_ID', string $mobile = '9999999999'): array
{
    $ch = curl_init();
    curl_setopt_array($ch, [
        CURLOPT_URL            => 'https://uat.kwikapi.com/api/v2/R-OFFER_check.php',
        CURLOPT_RETURNTRANSFER => true,
        CURLOPT_POST           => true,
        CURLOPT_POSTFIELDS     => http_build_query([
        'api_key' => $api_key,
        'opid' => $opid,
        'mobile' => $mobile,
        ]),
        CURLOPT_SSL_VERIFYPEER => true,
        CURLOPT_TIMEOUT        => 30,
        CURLOPT_HTTPHEADER     => ['Accept: application/json'],
    ]);

    $response = curl_exec($ch);
    $httpCode  = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    $curlError = curl_error($ch);
    curl_close($ch);

    if ($response === false) {
        throw new RuntimeException("cURL error: {$curlError}");
    }

    $data = json_decode($response, true);
    if (json_last_error() !== JSON_ERROR_NONE) {
        throw new RuntimeException('Invalid JSON response (HTTP ' . $httpCode . ')');
    }

    return $data;
}

// ── Example usage ─────────────────────────────────────────────────────────────
try {
    $result = kwik_r_offer_check('YOUR_API_KEY', 'OPERATOR_ID', '9999999999');

    if ($result['success'] ?? false) {
        echo "Success: " . ($result['message'] ?? 'OK') . PHP_EOL;
        echo json_encode($result, JSON_PRETTY_PRINT) . PHP_EOL;
    } else {
        echo "API Error: " . ($result['message'] ?? 'Unknown error') . PHP_EOL;
    }
} catch (RuntimeException $e) {
    echo "Request failed: " . $e->getMessage() . PHP_EOL;
    exit(1);
}
